core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -163,6 +163,28 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
   }
 
   /**
+   * Returns the top k (largest) elements for each key from this RDD as defined by the specified
+   * implicit Ordering[T].
+   * If the number of elements for a certain key is less than k, all of them will be returned.
+   *
+   * @param num k, the number of top elements to return
+   * @param ord the implicit ordering for T
+   * @return an RDD that contains the top k values for each key
+   */
+  def topByKey(num: Int)(implicit ord: Ordering[V]): RDD[(K, Array[V])] = {
+    aggregateByKey(new BoundedPriorityQueue[V](num)(ord))(
+      seqOp = (queue, item) => {
+        queue += item
+        queue
+      },
+      combOp = (queue1, queue2)  => {
+        queue1 ++= queue2
+        queue1
+      }
+    ).mapValues(_.toArray.sorted(ord.reverse))
```

Comment:
Does the `toArray` not already give you the top k in order? this seems to be the behavior already as it returns the array formed from the iterator in the underlying `PriorityQueue`. Worth testing I think. (That said, I suppose sorting an already-sorted array is pretty fast.)

Nit: a few lines above, an extra space in front of `=>`